include/onnxruntime/core/graph/graph.h

```diff
@@ -1139,16 +1139,40 @@
   const ONNX_NAMESPACE::GraphProto& ToGraphProto();
   ONNX_NAMESPACE::GraphProto ToGraphProto() const;
 
+  // Options to align external initializer offset.
+  struct OffsetAlignmentInfo {
+    // Offset will always be page aligned and allocation granularity aligned for mmap support.
+    // This is done by padding previous tensor data with zeros keeping same length.
+    bool align_offset = false;
+    // Alignment threshold for size of data.
+    // Having a low threshold will waste file space for small initializers.
+    // Only when tensor's data is > the page_align_threshold it will be force aligned.
+    // Default to 1MB.
```

Comment:
This needs to be re-phrased. tensor's data -> tensors's data size.

As a side not, I think if we mapped the whole file, then no individual alignment would be necessary except for the usual datatype alignment. We have initializers because we need them for model execution. All of them.

Reply:
will do, thanks.

For the side note, yeah, I think we can revisit them later, in theory we can map whole data file and save some disk space since we don't need to align per initializer. Current implementation is handle initializers one by one (load into memory, mmap or load into device) so this change will have quite large effort, we can do it that way if we expect enough benefit.